subprojects/core/src/test/groovy/org/gradle/initialization/DefaultProjectDescriptorTest.java

```diff
@@ -72,7 +72,7 @@ public void setName() {
         final ProjectDescriptorRegistry projectDescriptorRegistryMock = context.mock(ProjectDescriptorRegistry.class);
         projectDescriptor.setProjectDescriptorRegistry(projectDescriptorRegistryMock);
         context.checking(new Expectations() {{
-            one(projectDescriptorRegistryMock).changeDescriptorPath(Path.path(TEST_NAME), Path.path(Project.PATH_SEPARATOR + newName));
+            one(projectDescriptorRegistryMock).changeDescriptorPath(Path.path(Project.PATH_SEPARATOR + TEST_NAME), Path.path(Project.PATH_SEPARATOR + newName));
```

Comment:
This used to pass because the `equals` method of `Path` was broken and didn't take the `absolute` flag into account.